packages/eas-cli/src/build/validate.ts

```diff
@@ -50,3 +58,61 @@ export async function checkGoogleServicesFileAsync<T extends Platform>(
 function isInsideDirectory(file: string, directory: string): boolean {
   return file.startsWith(directory);
 }
+
+export async function validateIconForManagedProjectAsync<T extends Platform>(
+  ctx: CommonContext<T>
+): Promise<void> {
+  if (ctx.workflow !== Workflow.MANAGED) {
+    return;
+  }
+
+  if (ctx.platform === Platform.ANDROID) {
+    await validateAndroidIconAsync(ctx as CommonContext<Platform.ANDROID>);
+  } else {
+    await validateIosIconAsync(ctx as CommonContext<Platform.IOS>);
+  }
+}
+
+async function validateAndroidIconAsync(ctx: CommonContext<Platform.ANDROID>): Promise<void> {
+  if (!ctx.exp.android?.adaptiveIcon?.foregroundImage) {
+    return;
+  }
+
+  const { foregroundImage } = ctx.exp.android.adaptiveIcon;
+  if (!foregroundImage.endsWith('.png')) {
+    Log.error(`The Android Adaptive Icon foreground image must be a PNG file.`);
+    Log.error(`expo.android.adaptiveIcon.foregroundImage = ${chalk.bold(foregroundImage)}`);
+    Errors.exit(1);
+  }
```

Comment:
We are quickly checking the Android icon using `endsWith` method. Do we want to do the same for the iOS icon?